### PR TITLE
[ML] Fix logger declaration in ML plugins

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
This corrects what appears to have been a copy-paste error
where the logger for `MachineLearning` and `DataFrame` was wrongly
set to be that of `XPackPlugin`.

### CODE REVIEW

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/DataFrame.java [hunk 1]
```diff
@@ -94,7 +94,7 @@
     public static final Set<String> HEADER_FILTERS = new HashSet<>(
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/DataFrame.java]: @hendrikmuhs Also, this appears unused. Are there plans of using it?

Comment (droberts195) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/DataFrame.java]: I think it's been refactored out into `ClientHelper.SECURITY_HEADER_FILTERS` (if you search for usages of that you'll see it's used in the `dataframe` plugin), but the old duplicated variable has not been removed in this file.

**Review by droberts195 — APPROVED** [review]
LGTM

File: x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/DataFrame.java (hunk 1 above)
Comment (dimitris-athanasiou) [reply x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/DataFrame.java]: OK, I'll remove it then in this PR.

**Pull request merged**